File: core/src/main/java/vn/loitp/views/dialog/imersivedialog/ImmersiveDialogFragment.java
package vn.loitp.views.dialog.imersivedialog;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.WindowManager;

import vn.loitp.views.LToast;

/**
 * Created by LENOVO on 3/30/2018.
 */

public class ImmersiveDialogFragment extends DialogFragment {

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog alertDialog = new AlertDialog.Builder(getActivity())
                .setTitle("Example Dialog")
                .setMessage("Some text.")
                .create();

        // Temporarily set the dialogs window to not focusable to prevent the short
        // popup of the navigation bar.
        alertDialog.getWindow().addFlags(WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE);

        alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                LToast.show(getActivity(), "Touch OK");
            }
        });
        //int color = ContextCompat.getColor(getActivity(), R.color.colorPrimary);
        //alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setTextColor(color);
        return alertDialog;
    }

    public void showImmersive(Activity activity) {

        // Show the dialog.
        show(activity.getFragmentManager(), null);

        // It is necessary to call executePendingTransactions() on the FragmentManager
        // before hiding the navigation bar, because otherwise getWindow() would raise a
        // NullPointerException since the window was not yet created.
        getFragmentManager().executePendingTransactions();

        // Hide the navigation bar. It is important to do this after show() was called.
        // If we would do this in onCreateDialog(), we would get a requestFeature()
        // error.
        getDialog().getWindow().getDecorView().setSystemUiVisibility(
                getActivity().getWindow().getDecorView().getSystemUiVisibility()
        );

        // Make the dialogs window focusable again.
        getDialog().getWindow().clearFlags(
                WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE
        );

    }

}

File: core/src/main/java/com/views/dialog/imersivedialog/ImmersiveDialogFragment.kt
package com.views.dialog.imersivedialog

import android.app.Activity
import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.view.WindowManager
import androidx.fragment.app.DialogFragment
import com.core.base.BaseActivity
import com.core.utilities.LLog
import com.views.LToast

/**
 * Created by Loitp on 3/30/2018.
 */

class ImmersiveDialogFragment : DialogFragment() {
    private val TAG = javaClass.simpleName

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val alertDialog = AlertDialog.Builder(activity)
                .setTitle("Example Dialog")
                .setMessage("Some text.")
                .create()

        // Temporarily set the dialogs window to not focusable to prevent the short
        // popup of the navigation bar.
        alertDialog.window?.addFlags(WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE)

        alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "OK") { _, _ ->
            context?.let { c ->
                LToast.show(c, "Touch OK")
            }
        }
        //int color = ContextCompat.getColor(getActivity(), R.color.colorPrimary);
        //alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setTextColor(color);
        return alertDialog
    }

    fun showImmersive(activity: Activity) {
        if (activity is BaseActivity) {
            activity.supportFragmentManager.let { fm ->
                show(fm, TAG)
                LLog.d(TAG, "show")

                // It is necessary to call executePendingTransactions() on the FragmentManager
                // before hiding the navigation bar, because otherwise getWindow() would raise a
                // NullPointerException since the window was not yet created.
                fm.executePendingTransactions()

                // Hide the navigation bar. It is important to do this after show() was called.
                // If we would do this in onCreateDialog(), we would get a requestFeature()
                // error.
                dialog?.let { d ->
                    d.window?.decorView?.systemUiVisibility = activity.window.decorView.systemUiVisibility

                    // Make the dialogs window focusable again.
                    d.window?.clearFlags(WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE)
                }
            }
        }
    }
}